tools/send-to-caas/bulk-publish-to-caas.beta.js

```diff
@@ -0,0 +1,543 @@
+
+  SIGNEDIN.style.display = 'none';
+  SIGNEDOUT.style.display = 'none';
+  resetResultsTables();
+  if (successArr.length) {
+    showSuccessTable(successArr);
+  }
+  if (errorArr.length) {
+    showErrorTable(errorArr);
+  }
+
+  showAlert(`Successfully published ${successArr.length} pages. \n\n Failed to publish ${errorArr.length} pages.`);
+};
+
+const bulkPublish = async () => {
+  const accessToken = await checkIms();
+  if (!accessToken) return;
+
+  const { urls } = getConfig();
+
+  if (!urls) {
+    await showAlert('Enter a URL or list of URLs, each on a separate line, to be sent to CaaS.', { error: true });
+  }
+
+  const data = urls ? urls.split('\n') : '';
+
+  await processData(data, accessToken);
+};
+
+const loadFromLS = () => {
+  BODY.classList = '';
+  const ls = localStorage.getItem(LS_KEY);
+  if (!ls || !ls.includes('presetSelector')) return;
+  try {
+    setConfig(JSON.parse(ls));
+    const config = getConfig();
+    FIELDS.forEach((field) => {
+      document.getElementById(field).value = config[field] ?? DEFAULT_VALUES[field];
+      if (field === 'presetSelector' && config[field] === 'advanced') {
+        BODY.classList = 'advanced';
+      } else if (field === 'presetSelector' && config[field] !== 'default') {
+        BODY.classList = 'preset';
+      }
+    });
+    FIELDS_CB.forEach((field) => {
+      document.getElementById(field).checked = config[field] ?? DEFAULT_VALUES_CB[field];
+    });
+
+    /* c8 ignore next */
+  } catch (e) { /* do nothing */ }
+};
+
+const publishWarning = document.querySelector('.publish-warning');
+const checkCaasEnv = () => {
+  // eslint-disable-next-line no-undef
+  const caasEnvValue = caasEnvSelector.value || 'prod';
+  // eslint-disable-next-line no-undef
+  if (caasEnvValue === 'prod' && !draftOnly.checked) {
+    publishWarning.style.height = '30px';
+  } else {
+    publishWarning.style.height = '0';
+  }
+};
+
+// preset options
+const presetsJsonPath = 'https://milo.adobe.com/drafts/caas/bppresets.json';
+let presetsData = {};
+
+fetchExcelJson(presetsJsonPath).then((presets) => {
+  const separator = document.querySelector('.separator');
+  const parent = separator.parentElement;
+  presetsData = presets;
+  presets.forEach((preset) => {
+    const option = document.createElement('option');
+    option.value = preset.repo;
+    option.text = `${preset.name} (${preset.repo})`;
+    parent.insertBefore(option, separator);
+  });
+});
+
+const resetAdvancedOptions = () => {
+  /* eslint-disable no-undef */
+  caasEnvSelector.value = 'prod';
+  draftOnly.checked = false;
+  useHtml.checked = false;
+  usePreview.checked = false;
+  publishToFloodgate.value = 'default';
+  /* eslint-enable no-undef */
+};
+
+/* eslint-disable no-nested-ternary */
+const useDarkTheme = localStorage.getItem('bp-theme') === 'dark'
+  ? true
+  : localStorage.getItem('bp-theme') === 'light'
+    ? false
+    : window.matchMedia('(prefers-color-scheme: dark)').matches;
```

Comment:
Better to avoid nested ternary.  Something like:
```suggestion
const theme = localStorage.getItem('bp-theme');
const useDarkTheme = theme === 'dark' ||
  (theme !== 'light' && window.matchMedia('(prefers-color-scheme: dark)').matches);
```